sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -109,6 +112,16 @@ export interface ExponentialRetryPolicyOptions {
     retryDelayInMs?: number;
 }
 
+// @public
+export type FailoverHostGenerator = (retryState: RetryInformation, options: DefaultRetryPolicyOptions) => Generator<FailoverHostState | undefined, void, RetryInformation>;
```

Comment:
also curious if we need `FailoverHostState | undefined` since presumably they could return rather than yield to indicate there is no failover and we already have `TReturn` set to `void`

Reply:
The `undefined` union allows for the generator to skip a failover attempt based on the state of the current response without ruling out failover during future responses. For example, yielding `undefined` allows retrying against the same host a few times before failover. Returning from the generator would disallow any future failover attempts for that pipeline request. Will anyone actually make use of that behavior? Do we want to support it?

I'm actually curious what your thoughts are on setting `TReturn` to `never`. I can't think of any situations where one would need to return from the generator and stop future failover attempts. Except for the trivial case where the policy never attempts failover, but you can just not define the generator option.